src/components/form-elements/Select.jsx

```diff
@@ -75,13 +81,25 @@ export type SelectPropsType = {
    * @example <Select size="m" options={[{value: 'option1', text: 'Option1'},{value: 'option2', text: 'Select selector'}]} />
    */
   options?: Array<OptionsPropsType>,
+  /**
+   * Optional array of grouped options of the select
+   * @example <Select size="m" options={[{"label": "Label title","items": [{ "value": "option1", "text": "Option1" },{ "value": "option2", "text": "Select selector" }]}]
+} />
+   */
+  groupedOptions?: Array<GroupedOptionsPropsType>,
```

Comment:
Hey @ensisf , I think we need to figure out better interface than this. If we need to mimic native solution (and this is great opportunity to do it). We need to allow options to be grouped and without groups at the same time. 

I would be more to allow having `option` and `optgroup` as children of select but this would change completely your implementation. If we want to keep current way of passing options we need to extend current prop.

We could jump on a call if you you'd like to.